app/services/activities/import_from_csv.rb

```diff
@@ -125,12 +135,20 @@ def find_activity_by_roda_id(roda_id)
     class ActivityCreator
       attr_reader :errors, :row, :activity
 
-      def initialize(organisation, row)
-        @organisation = organisation
+      def initialize(user:, row:, delivery_partner_organisation:)
+        @user = user
+        @organisation = user.organisation
+        @delivery_partner_organisation = delivery_partner_organisation
         @errors = {}
         @row = row
         @converter = Converter.new(row)
         @parent_activity = fetch_parent(@row["Parent RODA ID"])
+
+        if @parent_activity && !ActivityPolicy.new(@user, @parent_activity).create_child?
```

Comment:
I wonder if `ActivityPolicy.create_child?` would benefit from `return false if record.nil?`, to remove the need to check it's not nil here (`if @parent_activity && ...`).

Reply:
I don't think we're able to do this, if the parent activity is nil we don't want to add an additional error to say the uploader is not authorised. But I agree it could be tidied up, not sure how best to though.